addons/xeh/CfgVehicles.hpp

```diff
@@ -546,4 +546,31 @@ class CfgVehicles {
     class Truck_02_aa_base_lxWS: Truck_02_base_F {
         XEH_ENABLED;
     };
+
+    // Spearhead 1944
+
+    class Tank;
+    class SPE_Armored_Target_Dummy: Tank {
+        class EventHandlers {
+            // This class has an empty CBA_Extended_EventHandlers, repopulate with actual event handlers
```

Comment:
Wait, what? Why?

Reply:
Good question, this is the SPE class structure
```cpp
class SPE_Armored_Target_Dummy: Tank
{
	class Eventhandlers
	{
		class CBA_Extended_EventHandlers{};
	};
};
```
Since the `XEH_ENABLED` macro adds the `EventHandlers` classes with inheritance from the `CBA_Extended_EventHandlers_base` class it causes a load order issue and UBC problem. This structure with macros usage avoids both problems.

Reply:
LOL. To fix, should put
```cpp
class SPE_Armored_Target_Dummy: Tank {
	class Eventhandlers {
		delete CBA_Extended_EventHandlers;
	};
};
```
here, and then
```cpp
    class Tank;
    class SPE_Armored_Target_Dummy: Tank {
        XEH_ENABLED;
    };
```
into [ee component](https://github.com/CBATeam/CBA_A3/blob/master/addons/ee/config.cpp).

Reply:
That would need a separate addon with cfgPatches `requiredAddons[] = { "Some_SPE_Addon" };` and either the new `skipWhenMissingDependencies` with 2.14 patch or `#if __has_include("some_spe_path")`

Reply:
Why? `delete` complains now if no class is there? Maybe add it in patch 1, then remove it in patch 2, then XEH_enabled in patch 3. XEH already has a ton of (now sub-) module duplicates for legacy reasons. If you need to change the requiredAddons for it, go ahead.

Your current solution is whack.

Reply:
I think https://github.com/CBATeam/CBA_A3/pull/1589 should help with this problem

Reply:
```
	class SPE_Armored_Target_Dummy: Tank
	{
		scope=1;
		class Eventhandlers
		{
			class CBA_Extended_EventHandlers
			{
			};
		};
```
it's scope=1 and nothing inherits from this

I'm guessing this is for something scripted,
they explicitly disabled eh and xeh on it
I think we might just want to trust them and not touch it